### ISSUE TITLE
Beta crash (fails to sync)

### ISSUE THREAD

**Issue by lunarcontrol** [nextcloud/android#8284]
The files seem to be stuck spinning (refreshing) forever and then the crashed page pops up randomly (even while in other apps) and by clicking on certain button like the 'media' button.

This only started happening but worked earlier.

Could this be because I have 2 A records for my external & internal ip addresses. I'm doing this because my router doesnt support hairpin NAT.

### Steps to reproduce
1. 
2. 
3. 

### Expected behaviour
App is able to sync correctly w/out crashing

### Actual behaviour
App gets stuck refreshing files and eventually crashes.

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

### Environment data
Android version: 9

Device model: SM-G950U

Stock or customized system:

Nextcloud app version: 3.16.0 RC2

Nextcloud server version: 21 latest stable

Reverse proxy:

#### Web server error log
```
I can get it if you would like.
```

#### Nextcloud log (data/nextcloud.log)
```
************ CAUSE OF ERROR ************

java.lang.NullPointerException
	at com.nextcloud.ui.ChooseAccountDialogFragment.getBinding(ChooseAccountDialogFragment.kt:71)
	at com.nextcloud.ui.ChooseAccountDialogFragment.avatarGenerated(ChooseAccountDialogFragment.kt:196)
	at com.owncloud.android.datamodel.ThumbnailsCacheManager$AvatarGenerationTask.onPostExecute(ThumbnailsCacheManager.java:921)
	at com.owncloud.android.datamodel.ThumbnailsCacheManager$AvatarGenerationTask.onPostExecute(ThumbnailsCacheManager.java:868)
	at android.os.AsyncTask.finish(AsyncTask.java:695)
	at android.os.AsyncTask.access$600(AsyncTask.java:180)
	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7073)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30160052
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamqltesq
Model: SM-G950U
Id: PPR1.180610.011
Product: dreamqltesq

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950USQU8DUA2

```
Second crash while filling out the github bug request:
```
************ CAUSE OF ERROR ************

java.lang.NullPointerException
	at com.nextcloud.ui.ChooseAccountDialogFragment.getBinding(ChooseAccountDialogFragment.kt:71)
	at com.nextcloud.ui.ChooseAccountDialogFragment.avatarGenerated(ChooseAccountDialogFragment.kt:196)
	at com.owncloud.android.datamodel.ThumbnailsCacheManager$AvatarGenerationTask.onPostExecute(ThumbnailsCacheManager.java:921)
	at com.owncloud.android.datamodel.ThumbnailsCacheManager$AvatarGenerationTask.onPostExecute(ThumbnailsCacheManager.java:868)
	at android.os.AsyncTask.finish(AsyncTask.java:695)
	at android.os.AsyncTask.access$600(AsyncTask.java:180)
	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7073)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30160052
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: samsung
Device: dreamqltesq
Model: SM-G950U
Id: PPR1.180610.011
Product: dreamqltesq

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: G950USQU8DUA2
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by lunarcontrol** [comment]
![image](https://user-images.githubusercontent.com/5004460/114944466-809ca100-9dfc-11eb-9cd4-f4bf24110a93.png)

I also do not have SSL configured

**Comment by tobiasKaminsky** [comment]
It seems that avatar generation fails and thus it tries to show default avatar, but meanwhile the UI/app is suspended and thus it crashes.
What is strange to me is, that you say it happens also in background.

This is about the choose account dialog, which is only shown if you click on it…

If you open up the app, wait for one minute, then suspend the app, does it then also crash?

**Comment by AndyScherzinger** [comment]
Fixed via #8356

**Comment by AndyScherzinger** [comment]
Duplicate of #8375
